### PR TITLE
Update the embargo form to use the multipart access rights

### LINKED ISSUE #3213: Update show page set embargo option to use multi-select access rights design
The two-menu design is described in #2967

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔
Fixes #3213

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (including writing to shared file systems or interacting with other SDR APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

⚡ ⚠ If this change updates the Argo UI, run all integration tests that use Argo and create a PR on the integration test repo to fix anything this change breaks. ⚡

### CODE REVIEW

File: spec/requests/embargo_spec.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@
       end
 
       it 'returns 403' do
-        patch "/items/#{druid}/embargo", params: { item: { embargo_release_date: '2100-01-01' } }
+        patch "/items/#{druid}/embargo", params: { embargo: { release_date: '2100-01-01' } }
```
Comment (edsu) [inline spec/requests/embargo_spec.rb:47]: 💚  this test!

File: app/components/edit/item/access_rights_component.rb [hunk 2]
```diff
@@ -3,13 +3,19 @@
...
         @form_builder = form_builder
+        @include_cdl = include_cdl
       end
 
       def f
         @form_builder
       end
+
+      # embargo form doesn't have CDL access
+      def include_cdl?
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/components/edit/item/access_rights_component.rb:16]: The convention for methods that return a boolean is to have the method end with a `?`, I don't think attr_reader will do that.

File: app/forms/embargo_form.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    allow_blank: true
+  }
+
+  validates :access_location, inclusion: {
+    in: %w[spec music ars art hoover m&m],
+    allow_blank: true
+  }
+
+  def self.model_name
+    ::ActiveModel::Name.new(nil, nil, 'Embargo')
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/forms/embargo_form.rb:27]: That means "starting at the root of the namespace tree"

File: app/forms/embargo_form.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  end
+
+  def id
+    model.externalIdentifier
+  end
+
+  # When the object is initialized, copy the properties from the cocina model to the form:
+  def setup_properties!(_options)
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/forms/embargo_form.rb:35]: Yes.

**Pull request merged**